src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/cache/CacheUtil.java

```diff
@@ -15,44 +15,64 @@
 
 package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cache;
 
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.Resource;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Metric;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jooq.Record;
+import org.jooq.Result;
 
 public class CacheUtil {
     private static final Logger LOG = LogManager.getLogger(CacheUtil.class);
 
     public static Double getTotalSizeInKB(final Metric cacheSizeGroupByOperation) {
-        double sizeTotalInKB = 0;
+        double totalSizeInKB = 0;
 
         if (cacheSizeGroupByOperation.getFlowUnits().size() > 0) {
             // we expect the Metric to have single flow unit since it is consumed locally
             MetricFlowUnit flowUnit = cacheSizeGroupByOperation.getFlowUnits().get(0);
             if (flowUnit.isEmpty() || flowUnit.getData() == null) {
-                return sizeTotalInKB;
+                return totalSizeInKB;
             }
 
-            // since the flow unit data is aggregated, we should have a single value
+            // since the flow unit data is aggregated by index, summing the size across indices
             if (flowUnit.getData().size() > 0) {
-                double size = flowUnit.getData().get(0).getValue(MetricsDB.SUM, Double.class);
-                if (Double.isNaN(size)) {
-                    LOG.error("Failed to parse metric in FlowUnit from {}", cacheSizeGroupByOperation.getClass().getName());
-                } else {
-                    sizeTotalInKB += size / 1024.0;
-                }
+                Result<Record> records = flowUnit.getData();
+                double size = records.stream().mapToDouble(
+                        record -> record.getValue(MetricsDB.SUM, Double.class)).sum();
+                totalSizeInKB += getSizeInKB(size);
             }
         }
-        return sizeTotalInKB;
+        return totalSizeInKB;
+    }
+
+    public static Double getSizeInKB(double sizeinBytes) {
+        double sizeInKB = 0;
+        if (Double.isNaN(sizeinBytes)) {
+            LOG.error("getSizeInKB called with NaN value");
```

Comment:
Can we throw an exception here instead? I try to avoid overloading return values with error codes in java. Off the top of my head, if cache size ever is less than 1KB, we cannot distinguish between that and error values.